### PR TITLE
vimPlugins.chadtree: add python dependencies

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

This PR is about two vim plugins from [ms-jpq](https://github.com/ms-jpq):
- [chadtree](https://github.com/ms-jpq/chadtree)
- [coq_nvim](https://github.com/ms-jpq/coq_nvim)

Both rely on the same python dependencies:
- pyyaml (which is already nicely packaged in `nixpkgs`)
- [pynvim_pp](https://github.com/ms-jpq/pynvim_pp)
- [std2](https://github.com/ms-jpq/std2)

In https://github.com/NixOS/nixpkgs/pull/223056, I packaged those two last python packages "inline" in the `overrides.nix` file.
The objective of this current PR is to also patch `vimPlugins.chadtree` to use those packaged dependencies instead of relying on a fragile runtime venv-based installation.

Here is the list of changes introduced:
- Package `std2` individually as `python3Packages.std2`
- Package `pynvim_pp` individually as `python3Packages.pynvim-pp`
- Change the override definition of `vimPlugins.coq_nvim` to use those two nix packages
- Rename `vimPlugins.chad` to `vimPlugins.chadtree`. (Do not worry, I also have set an alias `chad = chadtree`)
- Patch `vimPlugins.chadtree` in `overrides.nix` to use the nix packages for its python dependencies. The necessary code patch has also been included to ignore venv detection at runtime.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 241834` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>66 packages built:</summary>
  <ul>
    <li>python310Packages.pynvim-pp</li>
    <li>python310Packages.pynvim-pp.dist</li>
    <li>python310Packages.std2</li>
    <li>python310Packages.std2.dist</li>
    <li>python311Packages.pynvim-pp</li>
    <li>python311Packages.pynvim-pp.dist</li>
    <li>python311Packages.std2</li>
    <li>python311Packages.std2.dist</li>
    <li>vimPlugins.LeaderF</li>
    <li>vimPlugins.SchemaStore-nvim</li>
    <li>vimPlugins.SpaceVim</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vimPlugins.barbar-nvim</li>
    <li>vimPlugins.barbecue-nvim</li>
    <li>vimPlugins.catppuccin-nvim</li>
    <li>vimPlugins.chadtree</li>
    <li>vimPlugins.cmp-nvim-ultisnips</li>
    <li>vimPlugins.codeium-vim</li>
    <li>vimPlugins.completion-treesitter</li>
    <li>vimPlugins.coq-artifacts</li>
    <li>vimPlugins.coq-thirdparty</li>
    <li>vimPlugins.coq_nvim</li>
    <li>vimPlugins.deol-nvim</li>
    <li>vimPlugins.dropbar-nvim</li>
    <li>vimPlugins.elixir-tools-nvim</li>
    <li>vimPlugins.flash-nvim</li>
    <li>vimPlugins.friendly-snippets</li>
    <li>vimPlugins.fzf-lua</li>
    <li>vimPlugins.haskell-tools-nvim</li>
    <li>vimPlugins.heirline-nvim</li>
    <li>vimPlugins.indent-blankline-nvim</li>
    <li>vimPlugins.julia-vim</li>
    <li>vimPlugins.lazy-lsp-nvim</li>
    <li>vimPlugins.lean-nvim</li>
    <li>vimPlugins.leap-ast-nvim</li>
    <li>vimPlugins.luasnip</li>
    <li>vimPlugins.mason-lspconfig-nvim</li>
    <li>vimPlugins.neodev-nvim</li>
    <li>vimPlugins.neogit</li>
    <li>vimPlugins.nvim-highlite</li>
    <li>vimPlugins.nvim-lspconfig</li>
    <li>vimPlugins.nvim-navbuddy</li>
    <li>vimPlugins.nvim-navic</li>
    <li>vimPlugins.nvim-scrollview</li>
    <li>vimPlugins.nvim-treesitter</li>
    <li>vimPlugins.nvim-treesitter-parsers.cuda</li>
    <li>vimPlugins.nvim-treesitter-parsers.foam</li>
    <li>vimPlugins.nvim-treesitter-parsers.hlsl</li>
    <li>vimPlugins.nvim-treesitter-parsers.mermaid</li>
    <li>vimPlugins.nvim-treesitter-parsers.sql</li>
    <li>vimPlugins.nvim-treesitter-parsers.wing</li>
    <li>vimPlugins.nvim_context_vt</li>
    <li>vimPlugins.onedarkpro-nvim</li>
    <li>vimPlugins.orgmode</li>
    <li>vimPlugins.playground</li>
    <li>vimPlugins.refactoring-nvim</li>
    <li>vimPlugins.rest-nvim</li>
    <li>vimPlugins.satellite-nvim</li>
    <li>vimPlugins.ssr</li>
    <li>vimPlugins.toggleterm-nvim</li>
    <li>vimPlugins.tokyonight-nvim</li>
    <li>vimPlugins.unison</li>
    <li>vimPlugins.vifm-vim</li>
    <li>vimPlugins.vim-markbar</li>
    <li>vimPlugins.vim-nickel</li>
    <li>vimPlugins.vimtex</li>
  </ul>
</details>

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/development/python-modules/pynvim-pp/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    hash = "sha256-H1sCwU2wl9HO92LNkvkCb1iLbZrTNMmYA/8qy9uCgyU=";
+  };
+
+  propagatedBuildInputs = [
+    pynvim
+    setuptools
+  ];
+
+  doCheck = true;
+
```
Comment (figsoda) [inline pkgs/development/python-modules/pynvim-pp/default.nix:28]: I don't think that does anything
```suggestion
```
https://github.com/NixOS/nixpkgs/blob/3c7487575d9445185249a159046cc02ff364bff8/pkgs/development/interpreters/python/mk-python-derivation.nix#L218-L220

File: pkgs/development/python-modules/pynvim-pp/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    owner = "ms-jpq";
+    repo = "pynvim_pp";
+    rev = "e9ac3171c7edcc0be020d63840d6af4222b69540";
+    fetchSubmodules = false;
+    hash = "sha256-H1sCwU2wl9HO92LNkvkCb1iLbZrTNMmYA/8qy9uCgyU=";
+  };
+
+  propagatedBuildInputs = [
+    pynvim
+    setuptools
```
Comment (figsoda) [inline pkgs/development/python-modules/pynvim-pp/default.nix:24]: Is this required?

File: pkgs/development/python-modules/pynvim-pp/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  pname = "pynvim-pp";
+  version = "unstable-2023-07-05";
+
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "ms-jpq";
+    repo = "pynvim_pp";
+    rev = "e9ac3171c7edcc0be020d63840d6af4222b69540";
+    fetchSubmodules = false;
```
Comment (figsoda) [inline pkgs/development/python-modules/pynvim-pp/default.nix:18]:
```suggestion
```

File: pkgs/development/python-modules/std2/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  pname = "std2";
+  version = "unstable-2023-07-05";
+
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "ms-jpq";
+    repo = "std2";
+    rev = "1080b7b657f75352481808a906939d16b6d956a1";
+    fetchSubmodules = false;
```
Comment (figsoda) [inline pkgs/development/python-modules/std2/default.nix:17]:
```suggestion
```

File: pkgs/development/python-modules/std2/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    hash = "sha256-shAuCZvyFGrW26uTnNeyvlAIABb1d9oeKpk9hQbtuEQ=";
+  };
+
+  propagatedBuildInputs = [ setuptools ];
+
+  doCheck = true;
+
```
Comment (figsoda) [inline pkgs/development/python-modules/std2/default.nix:24]:
```suggestion
```

File: pkgs/development/python-modules/std2/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "ms-jpq";
+    repo = "std2";
+    rev = "1080b7b657f75352481808a906939d16b6d956a1";
+    fetchSubmodules = false;
+    hash = "sha256-shAuCZvyFGrW26uTnNeyvlAIABb1d9oeKpk9hQbtuEQ=";
+  };
+
+  propagatedBuildInputs = [ setuptools ];
```
Comment (figsoda) [inline pkgs/development/python-modules/std2/default.nix:21]: is this required?

Comment (GaetanLepage) [reply pkgs/development/python-modules/std2/default.nix:20]: Yes, it doesn't build without.
It only works with both `format = "pyproject"` and `propagatedBuildInputs = [ setuptools ];`.

File: pkgs/development/python-modules/pynvim-pp/default.nix (hunk 2 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/pynvim-pp/default.nix:23]: Yes, I tried without. Adding `setuptools` to `propagatedBuildInputs` seems to be common for `format = pyproject` packages.

Comment (figsoda) [reply pkgs/development/python-modules/pynvim-pp/default.nix:23]: works for me after moving setuptools to `nativeBuildIputs`

File: pkgs/development/python-modules/std2/default.nix (hunk 6 above)
Comment (figsoda) [reply pkgs/development/python-modules/std2/default.nix:20]: works for me after moving setuptools to `nativeBuildIputs`

Comment (GaetanLepage) [reply pkgs/development/python-modules/std2/default.nix]: Indeed ! I changed it in both packages.

**Comment by figsoda** [comment]
ofborg seems to be failing now due to what looks like outdated patches, could you take a look at that?

**Comment by GaetanLepage** [comment]
> ofborg seems to be failing now due to what looks like outdated patches, could you take a look at that?

Fixed !

**Review by figsoda — APPROVED** [review]

**Pull request merged**